Description: Allow impossible size disks
Author: Rick Balocca <rbalocca@vyatta.com>
Origin: https://bugzilla.vyatta.com/show_bug.cgi?id=3574

On some versions of VMware, the hypervisor emulation of the
BIOS reports bogus disk sizes. This stops Grub from booting.
The easy fix is to just ignore the BIOS reported disk size.
  
---
 disk/i386/pc/biosdisk.c |   11 -----------
 1 files changed, 0 insertions(+), 11 deletions(-)

--- a/disk/i386/pc/biosdisk.c	2011-04-26 18:32:46.695625502 -0700
+++ b/disk/i386/pc/biosdisk.c	2011-04-26 18:37:53.886958498 -0700
@@ -248,22 +248,11 @@ grub_biosdisk_rw (int cmd, grub_disk_t d
       unsigned coff, hoff, soff;
       unsigned head;
 
-      /* It is impossible to reach over 8064 MiB (a bit less than LBA24) with
-	 the traditional CHS access.  */
-      if (sector >
-	  1024 /* cylinders */ *
-	  256 /* heads */ *
-	  63 /* spt */)
-	return grub_error (GRUB_ERR_OUT_OF_RANGE, "%s out of disk", disk->name);
-
       soff = ((grub_uint32_t) sector) % data->sectors + 1;
       head = ((grub_uint32_t) sector) / data->sectors;
       hoff = head % data->heads;
       coff = head / data->heads;
 
-      if (coff >= data->cylinders)
-	return grub_error (GRUB_ERR_OUT_OF_RANGE, "%s out of disk", disk->name);
-
       if (grub_biosdisk_rw_standard (cmd + 0x02, data->drive,
 				     coff, hoff, soff, size, segment))
 	{
